### ISSUE TITLE
Refactoring the grammar for tabular environment

### ISSUE THREAD

**Issue by yudai-nkt** [area/language-latex#145]
I've been thinking that [the grammar for `tabular` environment](https://github.com/area/language-latex/blob/v1.1.1/grammars/latex.cson#L1527-L1590) was pretty messy, and I just found that @cedhuf is saying in #4

> I'll try to submit an 'ugly' patch to make 'tabular' at least usable by not adding those spaces (probably just by renaming the row)

Maybe we can sophisticate the grammar, I guess (and hope).

**Comment by marks1024** [comment]
I agree that it is a confusing part of the grammar.  Also the name choices do not highlight the column and row separators `&` and `\\` effectively (at least in a few syntax themes I checked).  Perhaps I can look into this issue.
